makefile

```diff
@@ -73,10 +50,42 @@ $(buildDir)/$(name): cmd/$(name)/$(name).go $(srcFiles)
 	$(gobin) build -o $@ $<
 # end cli targets
 
+# start output files
+testOutput := $(foreach target,$(testPackages),$(buildDir)/output.$(target).test)
+lintOutput := $(foreach target,$(lintPackages),$(buildDir)/output.$(target).lint)
+coverageOutput := $(foreach target,$(testPackages),$(buildDir)/output.$(target).coverage)
+coverageHtmlOutput := $(foreach target,$(testPackages),$(buildDir)/output.$(target).coverage.html)
+.PRECIOUS: $(coverageOutput) $(coverageHtmlOutput) $(lintOutput) $(testOutput)
+# end output files
+
+# start basic development targets
+compile: $(srcFiles)
+	$(gobin) build $(compilePackages)
+proto:
+	@mkdir -p remote/internal
+	protoc --go_out=plugins=grpc:remote/internal *.proto
+lint: $(lintOutput)
+test: $(testOutput)
+benchmark: $(buildDir)/run-benchmarks
+	./$<
+coverage: $(coverageOutput)
+coverage-html: $(coverageHtmlOutput)
+phony += compile lint test coverage coverage-html proto benchmarks
+
+# start convenience targets for running tests and coverage tasks on a
+# specific package.
+test-%: $(buildDir)/output.%.test
+	
+coverage-%: $(buildDir)/output.%.coverage
+	
+html-coverage-%: $(buildDir)/output.%.coverage.html
+	
+lint-%: $(buildDir)/output.%.lint
+	
```

Comment:
Can we remove these newlines?

Reply:
They're required so that make recognizes them as targets.